### ISSUE TITLE
Move repository from SSH to USB

### ISSUE THREAD

**Issue by MatejKovacic** [borgbase/vorta#253]
My first backup was to a USB disk, connected to my computer. Target directory was `/media/BorgBackup/mycomputer/`

After first backup, which took a little bit long time, I connected this USB disk to remote server. So my target is now ssh://user@xx.xx.xx.xxx/media/BorgBackup/mycomputer/

After first use, I got notification, that backup directory has been init as disk, not as remote ssh mountpoint (something like that, I don't remember exactly). But I was able to list all backups and I am doing backup normally.

However, internet connection is slow, so I am not backuping the whole machine. But every once in a month, I would like to backup the whole machine, and I would like to do it by connecting local disk to my computer, run full backup (all directories) and then connect USB disk back to remote server.

So, I did this, in Vorta I have two profiles ("local" and "remote"), local backup has been done, but when connecting USB disk back to remote server and clicking refresh, I cannot see backup point from "local" connection.

Any idea?

**Comment by m3nu** [comment]
When you change the location of a repo, Borg will ask for confirmation when first connecting. After that it will work fine.

Currently we don't catch this connection confirmation in Vorta, so the backup may not start. A workaround would be to manually "ping" (e.g. `borg info`) the repo at the new address and then use Vorta for everything else.

My strategy is to have different profiles. One backs up the whole machine to a USB drive. Another backs the main folders up to [BorgBase.com](https://www.borgbase.com). I run the first manually and the latter every hour in the background.

**Comment by MatejKovacic** [comment]
In terminal I say:
`borg list ssh://user@xx.xx.xx.xxx/media/BorgBackup/mycomputer/`

And can see listed all backups: 
```
Enter passphrase for key ssh://user@xx.xx.xx.xxx/media/BorgBackup/mycomputer: 
start_state                          Wed, 2019-03-13 22:08:17 [...]
mycomp-without-vbox-2019-04-02T18:37:59 Tue, 2019-04-02 18:38:08 [...]
mycomp-without-vbox-2019-04-11T12:43:28 Thu, 2019-04-11 12:44:03 [...]
mycomp-usbdisk-2019-04-15T08:50:55 Mon, 2019-04-15 08:50:55 [...]
mycomp-all-2019-04-15T09:22:58 Mon, 2019-04-15 09:23:11 [...]
```

Backup plans are:
`start_state` is fresh installation of a computer;
`mycomp-without-vbox` in backuping everything **except** virtual machines over ssh
`mycomp-usbdisk` is backuping everything (including virtual machines) directly to USB disk
`mycomp-all` is backuping everything (including virtual machines) over ssh

On a Vorta (archives tab) I can see:
`start_state`
`mycomp-without-vbox-2019-04-02`
`mycomp-without-vbox-2019-04-11`
`mycomp-all-2019-04-15`

I cannot see `mycomp-usbdisk-2019-04-15`, even if I click Refresh button.

So I assume Refresh button does not work correctly?

**Comment by m3nu** [comment]
Vorta simply parses the output from `borg list`. I'd be surprised if you are accessing the correct repo and something is missing.

Can you supply the relevant logs (*Misc > Log*) and possibly your settings database? (make sure there is no sensitive data or send it privately to hello@borgbase.com.

**Comment by MatejKovacic** [comment]
I just sent an e-mail. Is there a possibility, that the problem is, that the entry, which is not shown, contains underscore character _ ?

**Comment by m3nu** [comment]
Thanks. Nothing that stands out in your email. Can you also send the Settings database from `~/.config/Vorta/settings.sqlite`?

We don't use the archive ID as primary DB key, so adding multiple archives with the same ID should be possible.

Underscore shouldn't be an issue.

**Comment by m3nu** [comment]
I already see the issue. We are assuming that archive IDs are unique, which they aren't in your case. So it won't add them if they already exists in another repo.

I removed this constraint in PR #255. Please have a look if this solves the issue for you. This will make changes to your settings database and you shouldn't use the official version until this feature is released in version 0.6.18.

To install the test version: (or checkout the PR branch if you prefer)

```
$ pip uninstall vorta
$ pip install https://files.qmax.us/vorta-0.6.17-debug.tar.gz
```

Thanks for reporting and helping debug this.

**Comment by MatejKovacic** [comment]
Great, now it is working (howewer, it is `pip3`, not `pip`).

Can you post here info when new version will be available for install? I assume I would just need to do `sudo pip3 update vorta` then?

**Comment by m3nu** [comment]
Good. I'll do the update soon. Depending on other changes we have outstanding. Will ping you here.

I think the command in your case will be:

`$ sudo pip3 install -U vorta`

**Comment by MatejKovacic** [comment]
Is update already done?

**Comment by m3nu** [comment]
According to what I wrote in April – yes. For the details I would need to look into the previous posts.

> I removed this constraint in PR #255. Please have a look if this solves the issue for you. This will make changes to your settings database and you shouldn't use the official version until this feature is released in version 0.6.18.
